File: ts/src/special-cards/interest-engine.ts
import { SpecialCardDefinition } from './types';

export const interestEngineCard: SpecialCardDefinition = {
  id: 'interest-engine',
  name: 'Interest Engine',
  shortName: 'Engine',
  description: 'Passive: gain +3 extra interest after each cleared level.',
  rarity: 'uncommon',
  cost: 8,
  accent: '#76d6ff',
  modifyProjectedInterest: (projectedInterest) => projectedInterest + 3
};

File: godot/Scripts/SpecialCards/InterestEngine.cs
namespace Roshambo.SpecialCards;

public static class InterestEngine
{
	public static readonly SpecialCardDefinition Definition = new()
	{
		Id = "special:interest-engine",
		Name = "Interest Engine",
		ShortName = "Interest",
		Description = "Projected interest +3.",
		Rarity = SpecialCardRarity.Uncommon,
		Cost = 8,
		Accent = "#7cff8f",
		ModifyProjectedInterest = projected => projected + 3
	};
}

